### PR TITLE
initialize methods

### PR DESCRIPTION
**Pull request by matthuhiggins**
Make superstore more like active record

### CODE REVIEW

**Comment by kunruh9** [comment]
Hm, a lot more changes than I thought...I was thinking it would be modifications to `[instantiate](https://github.com/data-axle/superstore/blob/master/lib/superstore/persistence.rb#L22-L33)`

**Comment by matthuhiggins** [comment]
The only required change is the change to instantiate.

File: lib/superstore/persistence.rb [hunk 1]
```diff
@@ -24,7 +24,13 @@ def instantiate(attributes, column_types = {}, &block)
           object.instance_variable_set("@id", attributes['id']) if attributes['id']
           object.instance_variable_set("@new_record", false)
           object.instance_variable_set("@destroyed", false)
-          document = attributes['document'].is_a?(String) ? JSON.parse(attributes['document']) : attributes['document']
+
+          if attributes['document'].is_a?(String)
+            document = JSON.parse(attributes['document'])
+          else
+            document = attributes
+          end
```
[the remark this answers is not in the document]
Comment (bullfight) [reply lib/superstore/persistence.rb]: This looks ok to me, it provides the same interface provided by `initialize`

```ruby
Person.new( 'id' => 1234, 'first_name' => "foo")
```

File: lib/superstore/core.rb [hunk 2]
```diff
@@ -41,15 +41,14 @@ def initialize(attributes = nil)
...
+        init_internals
+
         @attributes = @attributes.deep_dup
         @attributes['created_at'] = nil
         @attributes['updated_at'] = nil
         @attributes.delete(self.class.primary_key)
         @id = nil
 
-        @new_record = true
-        @destroyed = false
```
[the remark this answers is not in the document]
Comment (matthuhiggins) [reply lib/superstore/core.rb]: Actually might be ok

**Comment by matthuhiggins** [comment]
ok I'm wrong. this is good.

**Pull request merged**